### DISCUSSION TITLE
ESP32-CAM : additional sensors?

### DISCUSSION THREAD

**Discussion by RSV-Tuono** [arendst/Tasmota#14207] [Q&A] [answered]
Hello Tasmota Specialists,

for my Project " Control and Monitor 3D Printer" over HomeAssistant I wanted to have a 
• wifi cam
• filament measuring unit with runout controller
• 4 relais switch to control light, van and vacuum cleaner
• End-Stop connector
• temperature monitoring for heating-bet and housing

I started with Raspi Zero and cam works ok, but with a large latency and some GPIO problems, so I checked out some other possibilities...
 ESP32-CAM comes next ... after 4 days of struggeling with templates, gpio settings, bad wifi connections and one damaged esp32 i got my new CAMs today, where one is damaged out of the box again. (Too much power consumption on usb) So flashing impossible ...
But: one pcs is runging after a few hours:
Flashing with ESP-Flasher works like a charm
• Template from here [https://templates.blakadder.com/ai-thinker_ESP32-CAM.html](url) still not working.
Can not see any stream in front view, Light Toggle works
• By now, firmware is actual http://ota.tasmota.com/tasmota32/tasmota32-webcam.bin 10.0.1.0.3 with ESP32 Dev Kit

Problem is to find the "counter" option in configuration view.
As soon as I change some of the free GPIOS, stream will stop and after reset HomeAssistant will not show up the videostream again.
I want to use this kind of sensors : https://www.amazon.de/s?k=B077X8XL56 with 3.3V

Question: what do I need to do to change the GPIO without losing camera stream?

Can someone help to make the Counter and EndSwitch work?
Any Help is highly appreciated ;) Thanks in advance

**Comment by arendst** [comment, marked as answer]
I hate to say but forget any sensor usage on an ESP32-webcam. The webcam code is heavy interrupt driven; any user interrupt will at least interrupt the stream (as you've noticed). Any longer sensor interrupt will interrogate the ESP32 watchdog resulting in a restart.

I've been struggling long to make the webcam work while saving config data to flash; without the stream halt a watchdog will occur.

So I'd say forget the ESP32-webcam for anything other than a simple camera stream.

**Comment by RSV-Tuono** [comment]
Mhh, not the answer i imagined :) , 
but thank you very much for the great work !

**Reply by sfromis** [reply]
Those boards are very cheap. You can get two, and only activate the camera on one, and have the other for sensors. Too much trouble cramming "too much" onto one tiny MCU.

**Comment by B4stl3r** [comment]
Hi, 
I'm struggling with the same.. Idea would be, to connect an PIR and a Touch Sensor and send these Events via MQTT, nothing more.

At least PIR is working well, when i add the touch sensor, I face NVS issues (settings wont be accepted / overwritten, resetting etc. is not possible anymore.

So should this in general be possible, cause then I would collect my findings and report an issue.

**Reply by Jason2866** [reply]
We will close any issue regarding webcam build when additional external hardware is connected.
When doing you walk alone and have to solve upcoming issues yourself.
Writing to NVS is very! problematic when camera stream is running. Problems ARE expected to arrive.
